### PR TITLE
HTTP Basic Authentication support. Close #8

### LINKED ISSUE #8: HTTP Basic Authentication support

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -19,18 +20,50 @@ type App struct {
 	Address     string
 	Port        string
 	PermitWrite bool
+	User        string
```
Comment (yudai) [inline app/app.go]: Auth is better than User as User is confusing like uid of the process.

**Comment by yudai** [comment]
Hi,
Thank you for  the PR. Could you replace the term `User` to `Auth` (or `Credential`)?

**Comment by mattn** [comment]
Which do you expect `gotty -c user:pass` or `gotty -a user:pass` ?

**Comment by yudai** [comment]
`-c` looks better to me.

**Comment by mattn** [comment]
will do. please wait.

**Comment by mattn** [comment]
done, please take a look.

**Comment by yudai** [comment]
Thanks! merged.

**Pull request merged**